spec/models/site_spec.rb

```diff
@@ -79,4 +79,47 @@
     subject { build(:site, location_name: 'Foo', code: '1') }
     its(:to_s) { should eq 'Foo (code: 1)' }
   end
+
+  describe '#copy_to_provider' do
+    let(:site) { build :site }
+    let(:provider) { create :provider, sites: [site] }
+    let(:recruitment_cycle) { find_or_create :recruitment_cycle }
+    let(:next_recruitment_cycle) { create :recruitment_cycle, :next }
+    let(:next_provider) {
+      create :provider,
+             sites: [],
+             provider_code: provider.provider_code,
+             recruitment_cycle: next_recruitment_cycle
+    }
+
+    it 'makes a copy of the course in the new provider' do
+      site.copy_to_provider(next_provider)
+
+      next_site = next_provider.reload.sites.find_by(code: site.code)
+      expect(next_site).not_to be_nil
+    end
+
+    it 'leaves the existing site alone' do
+      site.copy_to_provider(next_provider)
+
+      expect(provider.reload.sites).to eq [site]
+    end
+
+    context 'the site already exists in the new provider' do
+      let!(:next_site) {
+        create :site,
+               code: site.code,
+               provider: next_provider
+      }
+
+      it 'does not make a copy of the site' do
+        # Something strange is going on with sites ... setting the code as we
+        # do for next_site doesn't seem to work so we have to assign it here.
+        next_site.update code: site.code
```

Comment:
maybe along here... https://github.com/DFE-Digital/manage-courses-backend/blob/63ab44c1ee1f5b3567edbf9de0e1fee9728fab2c/app/models/site.rb#L29

Reply:
Ya, I saw that but I couldn't quite fit it in with the behaviour I was seeing, which is why I just dropped it. I'm tempted to leave it for now.